bindings/rust/build.rs

```diff
@@ -1,174 +1,72 @@
-#[cfg(feature = "build_unicorn_cmake")]
-use bytes::Buf;
-#[cfg(feature = "build_unicorn_cmake")]
-use flate2::read::GzDecoder;
 #[cfg(feature = "use_system_unicorn")]
 use pkg_config;
 #[cfg(feature = "build_unicorn_cmake")]
-use reqwest::header::USER_AGENT;
-#[cfg(feature = "build_unicorn_cmake")]
-use std::path::{Path, PathBuf};
-#[cfg(feature = "build_unicorn_cmake")]
-use std::{env, process::Command};
+use std::env;
 #[cfg(feature = "build_unicorn_cmake")]
-use tar::Archive;
-
+use std::path::Path;
 #[cfg(feature = "build_unicorn_cmake")]
-fn find_unicorn(unicorn_dir: &Path) -> Option<PathBuf> {
-    for entry in std::fs::read_dir(unicorn_dir).ok()? {
-        let entry = entry.unwrap();
-        let path = entry.path();
-
-        if path.is_dir() && path.file_name()?.to_str()?.contains("unicorn") {
-            return Some(path);
-        }
+use std::process::Command;
+
+#[cfg(all(feature = "build_unicorn_cmake"))]
+fn setup_env_msvc() {
+    let devenv_path = cc::windows_registry::find_tool(
+        env::var("TARGET").unwrap().as_str(), "devenv").unwrap();
+    let devenv_dir = devenv_path.path().parent().unwrap();
+
+    let cmake_path = devenv_dir.join(r"CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe");
+    let ninja_path = devenv_dir.join(r"CommonExtensions\Microsoft\CMake\Ninja\ninja.exe");
+    if !cmake_path.is_file() {
+        panic!("missing cmake");
     }
-
-    None
-}
-
-#[cfg(feature = "build_unicorn_cmake")]
-fn out_dir() -> PathBuf {
-    let out_dir = env::var("OUT_DIR").unwrap();
-    Path::new(&out_dir).to_path_buf()
-}
-
-#[cfg(feature = "build_unicorn_cmake")]
-fn download_unicorn() -> PathBuf {
```

Comment:
If we don't support downloading unicron tarball, users could only clone the full repo to use. I think it makes sense to allow users to use unicorn just by specifying the dependency in cargo.toml.

Reply:
Do you mean you want to avoid people having to download the git history? I don't think there's a good way around that (might be worth looking into), and suspect that's what everyone else does...

The current way is probably just as bad because of the amount of dependencies it brings in :) (try `cargo tree` in the unicorn rust dir!)

Reply:
> Do you mean you want to avoid people having to download the git history? I don't think there's a good way around that (might be worth looking into), and suspect that's what everyone else does...

No, users should have a way to use it as crate without cloning the repo. For example, just `cargo new hello` and add  unicorn to their cargo.toml (and get the bindings from crates.io) should work. Since at this moment rust still doesn't have a stable way to distribute binray so this is the only way I think.

Reply:
Sorry I'm really not a rust expert and my solution indeed seems dumb.

It looks like the correct way to handle this and you may work on it. Thanks for your patience to revise the rust bindings :).